integrations/access/servicenow/app.go

```diff
@@ -226,27 +232,54 @@ func (a *App) onWatcherEvent(ctx context.Context, event types.Event) error {
 }
 
 func (a *App) onPendingRequest(ctx context.Context, req types.AccessRequest) error {
-	if len(req.GetSystemAnnotations()) == 0 {
-		logger.Get(ctx).Debug("Cannot proceed further. Request is missing any annotations")
-		return nil
+	reqID := req.GetName()
+	log := logger.Get(ctx).WithField("reqId", reqID)
+
+	resourceNames, err := a.BaseApp.GetResourceNames(ctx, req)
+	if err != nil {
+		return trace.Wrap(err)
 	}
 
-	// First, try to create a notification incident.
-	isNew, notifyErr := a.tryNotifyService(ctx, req)
+	reqData := RequestData{
+		User:              req.GetUser(),
+		Roles:             req.GetRoles(),
+		RequestReason:     req.GetRequestReason(),
+		SystemAnnotations: req.GetSystemAnnotations(),
+		Resources:         resourceNames,
+	}
 
+	// Create plugin data if it didn't exist before.
+	isNew, err := a.modifyPluginData(ctx, reqID, func(existing *PluginData) (PluginData, bool) {
+		if existing != nil {
+			return PluginData{}, false
+		}
+		return PluginData{RequestData: reqData}, true
+	})
+	if err != nil {
+		return trace.Wrap(err, "updating plugin data")
+	}
+
+	if isNew {
+		log.Infof("Creating servicenow incident")
+		if err = a.createIncident(ctx, reqID, reqData); err != nil {
+			// Even if we failed to create the incident we try to auto-approve
+			return trace.NewAggregate(
+				trace.WrapWithMessage(err, "creating ServiceNow incident"),
+				trace.Wrap(a.tryApproveRequest(ctx, req)),
+			)
+		}
+	}
+	if reqReviews := req.GetReviews(); len(reqReviews) > 0 {
+		if err = a.postReviewNotes(ctx, reqID, reqReviews); err != nil {
+			return trace.Wrap(err, "posting review notes")
```

Comment:
I swear this is the last change 🙃 

```suggestion
			return trace.NewAggregate(
				trace.WrapWithMessage(err, "posting review notes"),
				trace.Wrap(a.tryApproveRequest(ctx, req)),
			)
```